tests/unit_tests/toporder/test_cam.py

```diff
@@ -0,0 +1,147 @@
+
+
+def test_given_dataset_and_rescaled_dataset_when_fitting_then_returns_equal_output():
+    X = dummy_sample(seed=seed)
+    model = CAM()
+    context = make_context().variables(observed=X.columns).build()
+    model.fit(X, context)
+    A = nx.to_numpy_array(model.graph_)
+    model.fit(X * 2, context)
+    A_rescaled = nx.to_numpy_array(model.graph_)
+    assert np.allclose(A, A_rescaled)
+
+
+def test_given_dataset_and_dataset_with_permuted_column_when_fitting_then_return_equal_outputs():
+    X = dummy_sample(seed=seed)
+    model = CAM()
+    context = make_context().variables(observed=X.columns).build()
+
+    # permute sample columns
+    permutation = [1, 3, 0, 2]
+    permuted_sample = X[permutation]  # permute pd.DataFrame columns
+
+    # Run inference on original and permuted data
+    model.fit(permuted_sample, context)
+    A_permuted = nx.to_numpy_array(model.graph_)
+    order_permuted = model.order_
+    model.fit(X, context)
+    A = nx.to_numpy_array(model.graph_)
+    order = model.order_
+
+    # Match variables order
+    back_permutation = [2, 0, 3, 1]
+    A_permuted = A_permuted[:, back_permutation]
+    A_permuted = A_permuted[back_permutation, :]
+
+    # permutation_order with correct variables name
+    permutation_dict = {k: p for k, p in enumerate(permutation)}
+    order_permuted = [permutation_dict[o] for o in order_permuted]
+    assert order_permuted == order
+    assert np.allclose(A_permuted, A)
+
+
+def test_given_adjacency_when_fitting_then_returns_dag_with_context_included_edges():
+    X = dummy_sample(seed=seed)
+    model = CAM()
+    context = make_context().variables(observed=X.columns).build()
+    model.fit(X, context)  # fit without context included edges
+    A = nx.to_numpy_array(model.graph_)
+    order = model.order_
+    A_dense = full_dag(order)
+    d = len(X.columns)
+    edges = []  # include all edges in A_dense and not in A
+    for i in range(d):
+        for j in range(d):
+            if A_dense[i, j] == 1 and A[i, j] == 0:
+                edges.append((i, j))
+    included_edges = nx.empty_graph(len(X.columns), create_using=nx.DiGraph)
+    included_edges.add_edges_from(edges)
+    context = make_context(context).edges(include=included_edges).build()
+    model.fit(X, context)  # fit with context included edges
+    A_included = nx.to_numpy_array(model.graph_)
+    assert np.allclose(A_dense, A_included)
+
+
+def test_given_adjacency_when_pruning_with_pns_then_returns_dag_with_context_included_edges():
+    X = dummy_sample(seed=seed)
+    model = CAM(pns=True)
+    G_dense = dummy_dense()
+    context_builder = make_context()
+    context = context_builder.variables(observed=X.columns).edges(include=G_dense).build()
+    model.fit(X, context)
+    A_included = nx.to_numpy_array(model.graph_)
+    A_dense = nx.to_numpy_array(G_dense)
+    assert np.allclose(A_dense, A_included)
+
+
+def test_given_custom_nodes_labels_when_fitting_then_input_output_labels_are_consistent():
```

Comment:
This is the unit test to check whether the labels are preserved in the inferred output graph